packages/peregrine/lib/store/reducers/app.js

```diff
@@ -39,6 +40,12 @@ const reducerMap = {
             isOnline: false,
             hasBeenOffline: true
         };
+    },
+    [actions.setPageLoading]: (state, { payload }) => {
```

Comment:
I'm apprehensive to add additional functionality to redux if we plan on removing it. I know we've talked about removal of `cart`, `checkout`, and `user` slices, but we haven't talked about the `app` slice, so I could be mistaken and we do intend to leave it around.

Let's talk about this PR in team time tomorrow, and just make sure we're on the same page. I would want to see this as a `@client` directive in Apollo, with resolvers for querying and mutating. The uncharted territory is if we want to continue using these patterns we're all used to in redux, and add actions and state to AppContext for convenience, or set the expectation that people understand how to query and mutate data with Apollo.

Reply:
Ah I should have mentioned this option to Dave yesterday. I didn't even consider client state :( I wouldn't want him to have to rewrite all of this but we should raise awareness about the possiblity of using Apollo's cache instead of redux. It's a great topic for team time tomorrow, hopefully everyone can make it :)

Reply:
I'll be there in the meeting! Sounds like we should make a decision and make sure it's documented and used in all similar features moving forward 😌

Reply:
I almost left a comment about this issue specifically, so thanks for raising it.

As of writing, the Redux store's `app` slice appears to be the best place we have to put this new piece of state. But like @tjwiebell said, our goal is to pare down the usage of our Redux store until we can safely eliminate it, so ordinarily we wouldn't want to expand it like we're doing in this PR.

Obviously though, I don't want to task @davemacaulay with doing the exploratory work of creating a parallel app state context, much less a replacement. However, if we feel like the Apollo-based approach will work in this case, and that it has enough examples elsewhere in the codebase that we could reasonably ask Dave to use it here, I'd be open to it.

Reply:
Happy to change this implementation if there is enough good other examples of how it works, let's discuss in developer team time later and work out a plan to move forward 🙂

Reply:
As discussed Geese will do a follow up story to explore how to use Apollo local cache instead of appContext for instances like this.